systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
+    private static final String CRUISE_CONTROL_POD_PREFIX = CLUSTER_NAME + "-cruise-control-";
+
+    private static final String CRUISE_CONTROL_CAPACITY_FILE_PATH = "/tmp/capacity.json";
+    private static final String CRUISE_CONTROL_CONFIGURATION_FILE_PATH = "/tmp/cruisecontrol.properties";
+
+    private static final String CRUISE_CONTROL_METRICS_TOPIC = "strimzi.cruisecontrol.metrics"; // partitions 1 , rf - 1
+    private static final String CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC = "strimzi.cruisecontrol.modeltrainingsamples"; // partitions 32 , rf - 2
+    private static final String CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC = "strimzi.cruisecontrol.partitionmetricsamples"; // partitions 32 , rf - 2
+
+    @Test
+    void testCruiseControlDeployment()  {
+        String ccStatusCommand = "curl -X GET localhost:" + CRUISE_CONTROL_DEFAULT_PORT + CRUISE_CONTROL_ENDPOINT;
+
+        String ccPodName = PodUtils.getFirstPodNameContaining("cruise-control");
+
+        String result = cmdKubeClient().execInPodContainer(ccPodName, "cruise-control",
+            "/bin/bash", "-c", ccStatusCommand).out();
+
+        LOGGER.info("Verifying that {} is running inside the Pod {} and REST API is available", CRUISE_CONTROL_NAME, ccPodName);
+
+        assertThat(result, not(containsString("404")));
+        assertThat(result, containsString("RUNNING"));
+
+        verifyThatCruiseControlTopicsArePresent();
+    }
+
+    @Test
+    void testUninstallingAndInstallationCruiseControl() {
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Removing Cruise Control to the classic Kafka.");
+            kafka.getSpec().setCruiseControl(null);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        LOGGER.info("Verifying that in {} is not present in the Kafka cluster", CRUISE_CONTROL_NAME);
+        assertThat(KafkaResource.kafkaClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getCruiseControl(), nullValue());
+
+        LOGGER.info("Verifying that {} pod is not present", CRUISE_CONTROL_NAME);
+        assertThat(kubeClient().listPodsByPrefixInName(CRUISE_CONTROL_POD_PREFIX).size(), is(0));
+
+        // TODO: should topic be deleted ? CURRENTLY CC let the topics in the kafka
+//        verifyThatCruiseControlTopicsAreNotPresent();
+
+        kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Adding Cruise Control to the classic Kafka.");
+            kafka.getSpec().setCruiseControl(new CruiseControlSpec());
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
```

Comment:
As opposite to the previous comment, could we verify that the metrics reporter is configured on Kafka?